### PR TITLE
chore(demo-playwright): rewrite combobox test to playwright

### PR DESCRIPTION
**Pull request by 40oleg**
## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Build or CI related changes
- [X] Tests related changes
- [ ] Documentation content changes

## What is the current behaviour?

[Closes # ](https://github.com/taiga-family/taiga-ui/issues/4912)

### CODE REVIEW

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts [hunk 1]
```diff
@@ -1,8 +1,10 @@
 import {TuiDocumentationPagePO, tuiGoto} from '@demo-playwright/utils';
-import {expect, test} from '@playwright/test';
+import {expect, Locator, Page, test} from '@playwright/test';
 
 const {describe} = test;
 
+test.use({testIdAttribute: `automation-id`});
```
Comment (40oleg) [inline projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:6]: I used this one to rename test attribute. Should I change name of test attributes right inside in the components or leave it like this?

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:6]: It is cool feature!
But we use Page Objects for our tests. They does not have direct access to `test.use`.

But I think you can add `testIdAttribute: "automation-id"` to this object (to enable it globally):
https://github.com/taiga-family/taiga-ui/blob/0a4bcf294896f0a5bc3064e593d74b545b12e126/projects/demo-playwright/playwright.config.ts#L25

And update all locators inside `/projects/demo-playwright/tests`:
Screenshot 2023-11-27 at 16 12 27

(but don't change `projects/demo-playwright/utils/page-objects` – they can published in future and should not depend on any global configs)

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts [hunk 2]
```diff
@@ -24,4 +26,122 @@ describe(`ComboBox`, () => {
...
+                page,
+            }) => {
+                const {apiPageExample} = new TuiDocumentationPagePO(page);
+
+                const {comboboxTextfield} = new TuiComboBoxPO(apiPageExample);
+
+                await visitBy(page, strict);
+                await openFormValue(apiPageExample);
+                await expect(page).toHaveScreenshot(
+                    `search-should-not-be-reset-strict-${strict}.png`,
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:42]: Too large snapshot

search-should-not-be-reset-strict-true-chromium-darwin

(+ other snapshots have the same problem)

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts [hunk 3]
```diff
@@ -24,4 +26,122 @@ describe(`ComboBox`, () => {
...
+    await locator.locator(`button span`).getByText(`Form value`).click();
+}
+
+async function focusWrapper(locator: Locator): Promise<void> {
+    await locator.click({force: true});
+}
+
+async function clickFirstOption(page: Page): Promise<void> {
+    await expect(page.getByTestId(`tui-data-list-wrapper__option`).first()).toBeVisible();
+    await page.getByTestId(`tui-data-list-wrapper__option`).first().click({force: true});
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:146]: `TuiComboBoxPO` extends `TuiTextfieldWithDataListPO`.

`TuiTextfieldWithDataListPO` already has method `selectOptions`.

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts [hunk 4]
```diff
@@ -24,4 +26,122 @@ describe(`ComboBox`, () => {
...
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Backspace`);
+
+                await page.waitForTimeout(200);
+
+                await comboboxTextfield.click();
+                await page.keyboard.press(`Backspace`);
+
+                await page.waitForTimeout(200);
```
Comment (nsbarsukov) [inline projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:94]: Don't repeat every code line from previous cypress test 😄 

Is this playwright test actually required this new explicit timeout ?

Drop it and use the following cli-command to locally check if test is flaky without it:

```
nx e2e demo-playwright combobox.spec.ts --repeat-each=50
```

**Review by nsbarsukov — CHANGES_REQUESTED** [review]

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts [hunk 5]
```diff
@@ -24,4 +26,122 @@ describe(`ComboBox`, () => {
...
+        });
+    });
 });
+
+async function visitBy(page: Page, strict: boolean): Promise<void> {
+    await tuiGoto(page, `components/combo-box/API?tuiMode=null&strict=${strict}`);
+}
+
+async function openFormValue(locator: Locator): Promise<void> {
+    await locator.locator(`button span`).getByText(`Form value`).click();
```
[the remark this answers is not in the document]
Comment (40oleg) [reply projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts]: fixed

File: projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts (hunk 2 above)
Comment (splincode) [reply projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:42]: @nsbarsukov friendly ping

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:42]: > May I just reduce viewport?

Just use `setViewportSize`, yep

Comment (nsbarsukov) [reply projects/demo-playwright/tests/kit/combo-box/combobox.spec.ts:42]: \+ rebase the branch, please (it contains merge conflicts with `main` 😢 )

**Comment by 40oleg** [comment]
Dont know that branch that was recreated cannot be used for reopening PR. And because of that PR was closed. I created a new one. https://github.com/taiga-family/taiga-ui/pull/6354

**Pull request closed without merging**